### PR TITLE
chore: upgrade dependencies

### PR DESCRIPTION
**Pull request by lauzadis**
Upgrade third party dependencies to their latest versions

## Issue \#

N/A

## Description of changes

The Kotlin SDK continues to aggressively take the latest versions of dependencies while we are in developer preview.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: ktlint-rules/src/main/kotlin/software/aws/ktlint/rules/CustomRuleSetProvider.kt [hunk 1]
```diff
@@ -1,13 +1,12 @@
...
-    override fun get() = RuleSet(
-        "custom-ktlint-rules",
-        CopyrightHeaderRule(),
-        ExpressionBodyRule(),
-        MultilineIfElseBlockRule(),
+class CustomRuleSetProvider : RuleSetProviderV2("custom-ktlint-rules", NO_ABOUT) {
+    override fun getRuleProviders() = setOf(
+        RuleProvider { CopyrightHeaderRule() },
+        RuleProvider { ExpressionBodyRule() },
+        RuleProvider { MultilineIfElseBlockRule()},
```
Comment (ianbotsf) [inline ktlint-rules/src/main/kotlin/software/aws/ktlint/rules/CustomRuleSetProvider.kt:10]: **Style**: Missing space before closing `}`. Shouldn't **ktlint** itself have caught this? 🤔

File: runtime/tracing/tracing-core/common/src/aws/smithy/kotlin/runtime/tracing/LoggingTraceProbe.kt [hunk 2]
```diff
@@ -14,7 +14,8 @@ import mu.KotlinLogging
 public object LoggingTraceProbe : TraceProbe {
     private fun EventLevel.loggerMethod(): (KLogger, () -> Any?) -> Unit = when (this) {
         EventLevel.Fatal,
-        EventLevel.Error, -> KLogger::error
+        EventLevel.Error,
+        -> KLogger::error
         EventLevel.Warning -> KLogger::warn
```
Comment (ianbotsf) [inline runtime/tracing/tracing-core/common/src/aws/smithy/kotlin/runtime/tracing/LoggingTraceProbe.kt:19]: **Comment**: Ugh, don't like the indentation on this one. (Not worth fighting though.)

**Review by ianbotsf — APPROVED** [review]

File: ktlint-rules/src/main/kotlin/software/aws/ktlint/rules/CopyrightHeaderRule.kt [hunk 3]
```diff
@@ -17,13 +12,14 @@ class CopyrightHeaderRule : Rule(
              */
         """.trimIndent()
     }
-    override fun visit(
+    override fun beforeVisitChildNodes(
```
[the remark this answers is not in the document]
Comment (lauzadis) [reply ktlint-rules/src/main/kotlin/software/aws/ktlint/rules/CopyrightHeaderRule.kt:15]: Yep, I checked each one

File: ktlint-rules/src/main/kotlin/software/aws/ktlint/rules/CustomRuleSetProvider.kt (hunk 1 above)
Comment (lauzadis) [reply ktlint-rules/src/main/kotlin/software/aws/ktlint/rules/CustomRuleSetProvider.kt:10]: It seems like the ktlint rules are not being run on ktlint configuration files, I will see if there's an easy way to enable that

**Pull request merged**